distribution/lib/Standard/Base/0.0.0-dev/src/Data/Json/Extensions.enso

```diff
@@ -0,0 +1,194 @@
+
+## UNSTABLE
+
+   Returns a display representation of the dataflow error on which it is called.
+
+   > Example
+     Displaying a dataflow error.
+
+         import Standard.Examples
+
+         example_display = Examples.throw_error.to_default_visualization_data
+Error.to_default_visualization_data : Text
+Error.to_default_visualization_data self = self.catch Any .to_default_visualization_data
+
+## Text to JSON conversion.
+
+   > Example
+     Convert the text "Hello World!" to JSON.
+
+         "Hello World!".to_json
+   > Example
+     Convert the text "cześć" to JSON.
+
+         "cześć".to_json
+Text.to_json : Json
+Text.to_json self = Json.String self
+
+## Method used by object builders to convert a value into a valid JSON key.
+
+   > Example
+     Ensure that the text "foo" is a JSON key.
+         "foo".to_json_key
+Text.to_json_key : Text
+Text.to_json_key self = self
+
+## Convert a boolean to JSON.
+
+   > Example
+     Convert `True` to JSON.
+         True.to_json
+Boolean.to_json : Json
+Boolean.to_json self = Json.Boolean self
+
+## Convert `Nothing` to JSON.
+
+   > Example
+     Convert `Nothing` to JSON.
+         Nothing.to_json
+Nothing.to_json : Json
+Nothing.to_json self = Json.Null
+
+
+## Number to JSON conversion.
+
+   > Example
+     Convert the number 8 to JSON.
+
+         8.to_json
+Number.to_json : Json
+Number.to_json self = Json.Number self
+
+## Vector to JSON conversion.
+
+   > Example
+     Convert a vector of numbers to JSON.
+
+         [1, 2, 3].to_json
+Vector.to_json : Json
+Vector.to_json self = Json.Array (self.map .to_json)
+
+## UNSTABLE
+
+   Transform the vector into text for displaying as part of its default
+   visualization.
+Vector.to_default_visualization_data : Text
+Vector.to_default_visualization_data self =
+    json = self.take 100 . to_json
+    json.to_text
+
+## UNSTABLE
+   ADVANCED
+
+   Returns a Text used to display this value in the IDE.
+
+   The particular representation is left unspecified and subject to change in
+   the future. The current implementation uses JSON serialization as the
+   default.
+
+   > Example
+     Converting an array to its default visualization representation.
+
+         [1, 2, 3, 4].to_array.to_default_visualization_data
+Array.to_default_visualization_data : Text
+Array.to_default_visualization_data self =
+    Vector.from_polyglot_array self . to_default_visualization_data
+
+## A Locale to Json conversion
+
+   > Example
+     Convert the default locale to JSON.
+
+         import Standard.Base.Data.Locale.Locale
+
+         example_to_json = Locale.default.to_json
+Locale.to_json : Json
+Locale.to_json self =
+    b = Vector.new_builder
+    b.append ["type", "Locale"]
+    if self.language.is_nothing.not then b.append ["language", self.language]
+    if self.country.is_nothing.not then b.append ["country", self.country]
+    if self.variant.is_nothing.not then b.append ["variant", self.variant]
+    Json.from_pairs b.to_vector
```

Comment:
I'm just not sure if this is the right place for this method?

While I appreciate decoupling the most primitive types to avoid dependency on Json, it does not seem to be the responsibility of the Json module to know how to serialize stuff. To me it seems that it is the `Locale` that should know how it is to be serialized to json.

Otherwise, will we have _all_ our `to_json` implementations placed in this one file? Seems bad to me.

I'm fine with keeping the most basic types here, as decoupling them from Json makes sense.
But for any other types, IMO it makes most sense for their `to_json` implementation to be next to such type, not in some different place. It even makes it harder to find it.
What do you think?

Reply:
Locale ended up used in a few core places so didn't want to wire Json into it - otherwise yes I agree.

Reply:
Okay, fair